### PR TITLE
Adding ME17B SDK changes.

### PR DESCRIPTION
**Pull request by Jacob-Scheiffler**
As the title says, this PR adds all the changes for the ME17B. 
The notable changes are:
- Adding another wakeup timer
- Adding an external I2S clock source
- Adding ERFO ready interrupts
- Adding ARM to RISCV interrupt
- Adding support for RTC power in low power modes when not in use
- Adding MXC_SYS_GetRevision function
- Various register updates (that didn't require source code updates)

### CODE REVIEW

File: Libraries/PeriphDrivers/Include/MAX32655/pt.h [hunk 1]
```diff
@@ -169,39 +169,116 @@ uint32_t MXC_PT_IsActive(uint32_t pts);
...
+ * 
+ * @note       This is just an explicit version of MXC_PT_EnableInt()
  *
  * @param      pts   Bit mask of which pulse trains to enable. Set the bit
  *                   position of each pulse train to enable it. Bit0-\>pt0,
  *                   Bit1-\>pt1... etc, 1 will enable the interrupt, 0 to leave
  *                   a PT channel in its current state.
  */
-void MXC_PT_EnableInt(uint32_t pts);
+void MXC_PT_EnableStopInt(uint32_t pts);
```
Comment (Jake-Carter) [inline Libraries/PeriphDrivers/Include/MAX32655/pt.h:181]: ME17 `Pulse_Train` example depends on `MXC_PT_EnableInt`.  Update example to use new functions to clear build warning

**Comment by Jake-Carter** [comment]
Strange that the actions are not showing for this PR, but there are quite a few build errors:

https://github.com/Analog-Devices-MSDK/msdk/actions/runs/4534175627/jobs/7988009241

File: Libraries/CMSIS/Device/Maxim/MAX32655/Source/GCC/startup_riscv_max32655.S [hunk 2]
```diff
@@ -112,14 +112,14 @@ __isr_vector:
   jal x0, TMR5_IRQHandlerWrap
   jal x0, I2C0_IRQHandlerWrap
   jal x0, UART0_IRQHandlerWrap
-  jal x0, RSV16_IRQHandlerWrap
+  jal x0, CM4_IRQHandlerWrap
```
[the remark this answers is not in the document]
Comment (Jacob-Scheiffler) [reply Libraries/CMSIS/Device/Maxim/MAX32655/Source/GCC/startup_riscv_max32655.S:115]: Not sure what you mean by "swapped"?

File: Libraries/CMSIS/Device/Maxim/MAX32655/Source/GCC/startup_max32655.S [hunk 3]
```diff
@@ -195,7 +195,7 @@ __isr_vector:
     .long BTLE_RX_AES_IRQHandler       /* 0x42  0x0108  66: BTLE RX AES Done */
     .long BTLE_INV_APB_ADDR_IRQHandler /* 0x43  0x010C  67: BTLE Invalid APB Address*/
     .long BTLE_IQ_DATA_VALID_IRQHandler/* 0x44  0x0110  68: BTLE IQ Data Valid */
-    .long WUT_IRQHandler               /* 0x45  0x0114  69: Wake up timer */
+    .long WUT0_IRQHandler               /* 0x45  0x0114  69: Wakeup Timer 0 */
```
[the remark this answers is not in the document]
Comment (Jacob-Scheiffler) [reply Libraries/CMSIS/Device/Maxim/MAX32655/Source/GCC/startup_max32655.S:198]: I added ```#define WUT_IRQHandler WUT0_IRQHandler``` to max32655.h, but I'm doubting whether or not that would work now. I'll re-test that 'cause I'm pretty sure I didn't test that change correctly.

[the remark this answers is not in the document]
Comment (Jacob-Scheiffler) [reply Libraries/CMSIS/Device/Maxim/MAX32655/Source/GCC/startup_max32655.S:198]: Re-tested and it works.

File: Libraries/CMSIS/Device/Maxim/MAX32655/Source/GCC/startup_riscv_max32655.S (hunk 2 above)
[the remark this answers is not in the document]
Comment (Jake-Carter) [reply Libraries/CMSIS/Device/Maxim/MAX32655/Source/GCC/startup_riscv_max32655.S:115]: In this file you've swapped the order that the RSCV16 and CM4 IRQs are laid out in the interrupt vector for the RISC-V core.  Some other stuff is swapped around too, just wondering if this was related to the Rev A vs Rev B functional differences or not

**Review by Jake-Carter — APPROVED** [review]

**Review by sihyung-maxim — APPROVED** [review]

**Pull request merged**